src/avram/migrator/drop_foreign_key_statement.cr

```diff
@@ -0,0 +1,22 @@
+# Generates an ALTER TABLE statement for dropping a foreign key constraint on a table.
+#
+# ### Usage
+#
+# ```
+# DropForeignKeyStatement.new(from: :comments, references: :users, column: :author_id).build
+# # => "ALTER TABLE comments DROP CONSTRAINT comments_author_id_fk;"
+# ```
+class Avram::Migrator::DropForeignKeyStatement
+  def initialize(@from : Symbol, @references : Symbol, @column : Symbol? = nil)
+  end
+
+  def build
+    foreign_key = @column || Wordsmith::Inflector.singularize(@references.to_s) + "_id"
+    String.build do |index|
+      index << "ALTER TABLE"
+      index << " #{@from}"
+      index << " DROP CONSTRAINT #{@from}_#{foreign_key}_fk"
```

Comment:
hmm, would it be better to have the optional value be the foreign key name? So, the foreign key would either be created from the reference or passed in explicitly? 🤔

Reply:
Is this what you mean? https://github.com/luckyframework/avram/pull/654/files#diff-ba436c31cb078c2767a29a28ec723092b8441934ea17e02cb97831099417ed60R10 There's that third arg that lets you specify the column it's on. Or do you mean just pass in the raw fk name itself? `name: "what_ever_i_want_fk"`

Reply:
Yea, I'm thinking that we provide an extra config option for this method but it doesn't actually support all the reasons for not being able to use the default flow

Reply:
I was thinking about this over the weekend. The CreateForeignKey class doesn't give you the option to do a raw name https://github.com/luckyframework/avram/blob/master/src/avram/migrator/create_foreign_key_statement.cr#L10. I wonder if we should say at that level, you just need to run the `execute` yourself. These classes are more like the midway between the "lucky way" (using helper methods), and the escape hatch (writing yourself). 

If we think these should open up the options all the way, then we will want to make sure they all follow those same rules. What are your thoughts?

Reply:
I'm fine with this thinking 👍